usefull_app/user_api/tests/test_passengers_rate.py

```diff
@@ -0,0 +1,398 @@
+
+
+class PostPassengersRateTestCase(TestMixin, TransactionTestCase):
+    """
+    Tests for /passengers/{id}/rate API endpoint.
+    """
+    def setUp(self) -> None:
+        """
+        Test case initialization method.
+        """
+        self.client = APIClient()
+
+    def test_post_passengers_rate_no_test_rate_data_in_db_self_rate(self) -> None:
+        """
+        Test POST /passengers/{id}/rate
+        [X] user data added to db
+        [X] passenger data added to db
+        [ ] no rate data added to db
+        passenger rating himself.
+        """
+        passenger = self.add_passenger_to_db()
+        url = reverse('passenger-ratings-list', args=[f'{passenger.id}'])
+        payload_data = copy.deepcopy(request_test_passenger_rate_data.ADD_PASSENGER_RATE_TEST_DATA)
+        payload_data['user'] = passenger.user_id
+        response = self.client.post(url, data=payload_data, format='json')
+        expected_result = response_test_passenger_rate_data.RESPONSE_PASSENGER_SELF_RATE_ERR_MSG
+        self.assertEqual(response.json(), expected_result)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(PassengerRating.objects.count(), 0)
+
+    def test_post_passengers_rate_no_test_rate_data_db(self) -> None:
+        """
+        Test POST /passengers/{id}/rate
+        [X] user data added to db
+        [X] passenger data added to db
+        [ ] no rate data added to db
+        valid rate test case.
+        """
+        # rate passenger
+        passenger = self.add_passenger_to_db()
+        # rated by user
+        user = self.add_random_users_to_db()
+        url = reverse('passenger-ratings-list', args=[f'{passenger.id}'])
+        payload_data = copy.deepcopy(request_test_passenger_rate_data.ADD_PASSENGER_RATE_TEST_DATA)
+        payload_data['user'] = user.id
+        response = self.client.post(url, data=payload_data, format='json')
+        expected_result = response_test_passenger_rate_data.RESPONSE_PASSENGER_RATE_TEST_DATA
+        self.assertEqual(response.json(), expected_result)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(PassengerRating.objects.count(), 1)
+
+    def test_post_passengers_rate_add_duplicate_rate(self) -> None:
+        """
+        Test POST /passengers/{id}/rate
+        [X] user data added to db
+        [X] passenger data added to db
+        [X] rate data added to db
+        check for handling duplicate rate in db.
+        """
+        # rate passenger
+        passenger = self.add_passenger_to_db()
+        # rated by user
+        user = self.add_random_users_to_db()
+        url = reverse('passenger-ratings-list', args=[f'{passenger.id}'])
+        payload_data = copy.deepcopy(request_test_passenger_rate_data.ADD_PASSENGER_RATE_TEST_DATA)
+        payload_data['user'] = user.id
+        # first rate
+        self.client.post(url, data=payload_data, format='json')
+        # duplicate
```

Comment:
Please make comments like here more verbose.
```suggestion
        # Try to rate passenger again with same payload.
```

Reply:
More verbose comments added in new commit.